components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -2676,6 +2693,140 @@ export function createWalletApi () {
+            } = baseQuery(undefined)
+            const priceHistory = await assetRatioService.getPriceHistory(arg.tokenParam, arg.vsAsset, arg.timeFrame)
+            return {
+              data: priceHistory.values
+            }
+          } catch(error) {
+            console.error('Error getting price history: ', error)
+            throw error
+          }
+        },
+        providesTags: (_result, err, arg) => err
+        ? ['PriceHistory']
+        : [{ type: 'PriceHistory', id: `${arg.tokenParam}-${arg.vsAsset}-${arg.timeFrame}` }]
+      }),
+      getNftMetadata: query<NFTMetadataReturnType, BraveWallet.BlockchainToken>({
+        queryFn: async (arg, _api, _extraOptions, baseQuery) => {
+          try {
+            const { data: api } = baseQuery(undefined)
+            const { jsonRpcService } = api
+            const result = arg.coin === BraveWallet.CoinType.ETH
+              ? await jsonRpcService.getERC721Metadata(arg.contractAddress, arg.tokenId, arg.chainId)
+              : arg.coin === BraveWallet.CoinType.SOL
+                ? await jsonRpcService.getSolTokenMetadata(arg.chainId, arg.contractAddress)
+                : undefined
+              
+            if (!result?.error) {
+              const response = result?.response && JSON.parse(result.response)
+              const attributes = Array.isArray(response.attributes)
+              ? response.attributes.map((attr: { trait_type: string; value: string }) => ({ traitType: attr.trait_type, value: attr.value }))
+              : []
+              const tokenNetwork = await getNetwork(api, arg)
+              const nftMetadata: NFTMetadataReturnType = {
+                metadataUrl: result?.tokenUrl || '',
+                chainName: tokenNetwork?.chainName || '',
+                tokenType:
+                  arg.coin === BraveWallet.CoinType.ETH
+                    ? 'ERC721'
+                    : arg.coin === BraveWallet.CoinType.SOL
+                    ? 'SPL'
+                    : '',
+                tokenID: arg.tokenId,
+                imageURL: response.image.startsWith('data:image/')
+                  ? response.image
+                  : await translateToNftGateway(response.image),
+                imageMimeType: 'image/*',
+                floorFiatPrice: '',
+                floorCryptoPrice: '',
+                contractInformation: {
+                  address: arg.contractAddress,
+                  name: response.name,
+                  description: response.description,
+                  website: '',
+                  facebook: '',
+                  logo: '',
+                  twitter: ''
+                },
+                attributes
+              }
+
+              return {
+                data: nftMetadata
+              }
+            } else {
+              console.error('Error fetching NFT metadata: ', result.error, result.errorMessage)
+              throw new Error(result.errorMessage)
```

Comment:
no need to log here. the "catch" below will log the error

```suggestion
              throw new Error(result.errorMessage)
```